### DISCUSSION TITLE
Running locally from sources

### DISCUSSION THREAD

**Discussion by Mark-Giesen** [WeblateOrg/weblate#9978] [Q&A] [answered]
This weekend I finally got to a retry of contributing to development from my Windows machine. Since the docs state the best experience is installing from sources I tried this.

A previous attempt failed, because not all worked on windows, but I realized it should be easy using WSL2. Well, that seems to be true. I've come very close to a good test run, but not quite yet. For me it's hard to identify what I did wrong or missed, maybe someone in this community can easily see what's wrong? I've included the output from "manage.py test" command. 

I assume a dot means a test passed, an s probably skipped, but is that OK? An E probably means Error, that is probably NOT OK. Some of these have some more info like "Can not guess type of plural for cs", this does not seem an installation error to me.
Than there are some stacktraces about a missing directory 'msgmerge' or a module 'responses'. 

Maybe all related to some setting I missed?

[test_results_weblate.txt](https://github.com/WeblateOrg/weblate/files/12648794/test_results_weblate.txt)

**Comment by nijel** [comment]
From a quick look, you are missing gettext-tools and installed test depencies. I didn't read the full log, though.

**Comment by Mark-Giesen** [comment]
Thanks for the quick answer, shouldn't those have been included in one of the steps in the docs? If so, do you know which one? A step like "pip install -e 'weblate-src[all]'" looks like a candidate to me ;-).

**Reply by Mark-Giesen** [reply]
There is a requirements-dev.txt files, which I could pip, but I assumed this was done by the pip install I mentioned here?

**Reply by nijel** [reply]
These are only runtime deps. I've clarified that in 673aba663a.

**Reply by Mark-Giesen** [reply]
Ouch, user error. I guess I was blinded by the "all" parameter. I'll give it an other go. Thanks.

**Reply by Mark-Giesen** [reply]
Ah, I see what went wrong, I started (as asked) with the [Debian and Ubuntu](https://docs.weblate.org/en/latest/admin/install/venv-debian.html) page and there is this command: pip install "Weblate[all]"; without the -e. I should have stopped before that line instead of after that line. But this command worked yesterday!
The command with the -e parameter gives an error though:

> ERROR: weblate-src[all,test] is not a valid editable requirement. It should either be a path to a local project or a VCS URL (beginning with bzr+http, bzr+https, bzr+ssh, bzr+sftp, bzr+ftp, bzr+lp, bzr+file, git+http, git+https, git+ssh, git+git, git+file, hg+file, hg+http, hg+https, hg+ssh, hg+static-http, svn+ssh, svn+http, svn+https, svn+svn, svn+file).

trying without the -e parameter I get an error as well now:

> ERROR: Could not find a version that satisfies the requirement weblate-src[all,test] (from versions: none)
> ERROR: No matching distribution found for weblate-src[all,test]

**Reply by nijel** [reply]
`-e` requires path to install from. `weblate-src` is checked out in step 2 at https://docs.weblate.org/en/latest/admin/install/source.html

**Reply by Mark-Giesen** [reply]
Thanks, this triggered one more try. Apparently I skipped one option: starting this command with de -e parameter in the parentfolder of weblate-src. It ran and I started the test again: fingers crossed.

**Reply by nijel** [reply]
You can also use `pip install -e .[all,test]` if you are in folder with Weblate sources.

**Reply by Mark-Giesen** [reply]
I get that now. I was looking for the error in the installation guide, but it ended up being an user error ;-). At the top of "installing from sources" is clearly states "...up to installing Weblate". I think this means until step 3 of "Python modules", but I understood as "until (including) the weblate install".
I still get some errors during test, including the msgmerge ones, zo I guess the gettext is not included in the install of [all,test]. I'll post the results shortly.

**Reply by nijel** [reply]
gettext needs to be installed on the system, it was apaprently missing from the instructions (because it is only needed for the msgmerge add-on and is detected at runtime). I've added that in 9909bc4a61

**Reply by Mark-Giesen** [reply]
Great, thanks! I've installed it an will run the tests again tonight.

**Reply by Mark-Giesen** [reply]
Testresult at end states OK, but output mentions some errors still, maybe these are supposed to happen and are part of tests. Some "Transalations not found" errors as wel.

I'll go with the OK at the end for now.

[test_log.txt](https://github.com/WeblateOrg/weblate/files/12705741/test_log.txt)

**Reply by nijel** [reply, marked as answer]
You can ignore the output when tests pass, unfortunately Django test runner doesn't capture output by default (unlike pytest), so that makes running some tests unnecessarily noisy.
